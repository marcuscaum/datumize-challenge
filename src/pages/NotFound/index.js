import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <section>
    <h1>404</h1>
    <h2>Page not found</h2>
    <Link to="/projects">Go to projects</Link>
  </section>
);

export default NotFound;
