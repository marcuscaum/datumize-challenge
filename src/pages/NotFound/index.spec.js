import React from 'react';
import { shallow } from 'enzyme';

import NotFound from './index';

const setup = props => shallow(<NotFound {...props} />);

describe('[Page] NotFound', () => {
  it('should render the page', () => {
    const wrapper = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
