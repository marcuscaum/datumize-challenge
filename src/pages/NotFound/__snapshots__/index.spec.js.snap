// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] NotFound should render the page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          404
</h1>,
        <h2>
          Page not found
</h2>,
        <Link
          replace={false}
          to="/projects"
>
          Go to projects
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404",
        },
        "ref": null,
        "rendered": "404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page not found",
        },
        "ref": null,
        "rendered": "Page not found",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Go to projects",
          "replace": false,
          "to": "/projects",
        },
        "ref": null,
        "rendered": "Go to projects",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            404
</h1>,
          <h2>
            Page not found
</h2>,
          <Link
            replace={false}
            to="/projects"
>
            Go to projects
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404",
          },
          "ref": null,
          "rendered": "404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page not found",
          },
          "ref": null,
          "rendered": "Page not found",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Go to projects",
            "replace": false,
            "to": "/projects",
          },
          "ref": null,
          "rendered": "Go to projects",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
